### PR TITLE
add what's new in tidb 4.0

### PR DESCRIPTION
**Pull request by TomShawn**
### What is changed, added or deleted? (Required)

add what's new in tidb 4.0.

### Which TiDB version(s) do your changes apply to? (Required)

- [x] master (the latest development version)
- [x] v4.0 (TiDB 4.0 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### CODE REVIEW

File: whats-new-in-tidb-4.0.md [hunk 1]
```diff
@@ -0,0 +1,96 @@
+---
+title: What's New in TiDB 4.0
+summary: 了解 TiDB 4.0 版本的新特性。
+category: introduction
+---
+
+# What's New in TiDB 4.0
+
+随着 TiDB v4.0 于 2020 年 5 月 28 日正式发布，TiDB v4.0 在稳定性、易用性、性能、安全和功能方面进行了大量的改进。本文概括性地介绍改进的内容，用户可根据实际情况决定升级到 TiDB v4.0。 本文仅总结简绍 v4.0 中面向用户且最重要的改进事项，功能及错误修复的完整列表，请参阅早期的 Release Notes。
```
Comment (lilin90) [inline whats-new-in-tidb-4.0.md:9]: - 去掉“简绍”？
- 去掉“早期”
- 给 "Release Notes" 添加 4.0 rn 的链接

Comment (TomShawn) [reply whats-new-in-tidb-4.0.md]: Done

File: whats-new-in-tidb-4.0.md [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
++ 支持大事务，最大事务限制由 100 MB 提升到了 10 GB，同时支持乐观事务和悲观事务。详情参阅：[事务限制](/transaction-overview.md#事务限制)。
+
+### SQL 功能
+
+- 在 SQL Plan Management 中引入了 SQL Bind 的自动捕获和演进，提升易用性和执行计划稳定性。详情参阅：[绑定执行计划](/sql-plan-management.md)。
+- 新增 15 种 SQL Hint 用于控制优化器生成执行计划，和执行引擎执行查询时的行为。详情参阅：[SQL Hint](/optimizer-hints.md)。
+- 支持 `SELECT INTO outfile` 语句，该语句用来将表数据导出到指定的文本文件中，配合上 `LOAD DATA`，可以方便的在数据库之间导入/导出数据。
+- 支持自定义序列对象 Sequence，提供 `CACHE/NO_CACHE`、`CYCLE/NO_CYCLE` 选项定义序列的不同特性，满足序列生成的各种需求，用户可以通过 Sequence 替代第三方 ID 生成服务。详情参阅：[Sequence](/sql-statements/sql-statement-create-sequence.md)。
+- 新增 `Flashback` 命令，支持恢复被 `Truncate` 的表。详情参阅：[`Flashback`](/sql-statements/sql-statement-flashback-table.md)。
+- 新增查询数据时将 Join、Sort 中间结果写入本地磁盘，防止查询语句占用内存过多导致系统 OOM 的问题，提升系统的稳定性。。
```
Comment (yikeke) [inline whats-new-in-tidb-4.0.md:56]:
```suggestion
- 新增查询数据时将 Join、Sort 中间结果写入本地磁盘，防止查询语句占用内存过多导致系统 OOM 的问题，提升系统的稳定性。
```

File: whats-new-in-tidb-4.0.md [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+
+### 部署运维工具
+
+TiUP 是 4.0 版本中新推出的包管理器的工具，主要用于管理 TiDB 生态内的所有的包，提供组件管理、Playground、Cluster、TUF、离线部署等功能，将安装、部署、运维 TiDB 工具化，提升 DBA 部署、运维 TiDB 的效率。详情参阅：[TiUP](/tiup/tiup-overview.md)，具体的功能如下：
+
+- 组件管理功能，提供一键式组件信息查询、安装、升级、卸载等功能，方便 DBA 管理 TiDB 的所有组件。
+- 集群管理功能 (Cluster)：提供一键式 TiDB 集群的部署、运维 TiDB 功能，包括：安装、部署、扩容、缩容、升级、配置变更、启动、停止、重启，查询集群状信息等，支持管理多个 TiDB 集群。
+- 本地部署功能 (Playground): 提供快速在本地部署一个 TiDB 集群，快速体验、了解 TiDB 的基本功能，注意：此功能仅用于快速了解 TiDB，不适合上生产。
+- 私有镜像管理 (Mirror): 当无法通过公网访问 TiUP 官方镜像时，TiUP 提供构建私有镜像的方案，帮助用户构建私有镜像及提供离线部署部署的功能。
+- 性能测试功能 (Benchmark): 提供一键部署性能测试工具的功能，主要提供 TPC-C、TPC-H 两种性能测试的 workload 。
```
Comment (yikeke) [inline whats-new-in-tidb-4.0.md:42]:
```suggestion
- 性能测试功能 (Benchmark): 提供一键部署性能测试工具的功能，主要提供 TPC-C、TPC-H 两种性能测试的 workload。
```

**Pull request merged**